src/util/mdb-exportjson.c

```diff
@@ -0,0 +1,186 @@
+/* MDB Tools - A library for reading MS Access database file
+ * Copyright (C) 2000 Brian Bruns
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "mdbtools.h"
+
+#ifdef DMALLOC
+#include "dmalloc.h"
+#endif
+
+#undef MDB_BIND_SIZE
+#define MDB_BIND_SIZE 200000
+
+#define is_quote_type(x) (x==MDB_TEXT || x==MDB_OLE || x==MDB_MEMO || x==MDB_DATETIME || x==MDB_BINARY || x==MDB_REPID)
+#define is_binary_type(x) (x==MDB_OLE || x==MDB_BINARY || x==MDB_REPID)
+
+static char *quote_char = "\"";
+static char *escape_char = "\\";
+static char *separator_char = ":";
+static char *row_start = "{";
+static char *row_end = "}\n";
+static char *delimiter = ",";
+static size_t quote_len = 1; //strlen(quote_char); /* multibyte */
+static size_t orig_escape_len = 1; //strlen(escape_char);
+static int drop_nonascii;
+
+
+//#define DONT_ESCAPE_ESCAPE
+static void
+print_quoted_value(FILE *outfile, char* value, int bin_len) {
+	fputs(quote_char, outfile);
+	int is_binary = (bin_len != -1);
+	while (1) {
+		if (is_binary) {
+			if (!bin_len--)
+				break;
+		} else /* use \0 sentry */
+			if (!*value)
+				break;
+
+		if (quote_len && !strncmp(value, quote_char, quote_len)) {
+			fprintf(outfile, "%s%s", escape_char, quote_char);
+			value += quote_len;
+#ifndef DONT_ESCAPE_ESCAPE
+		} else if (orig_escape_len && !strncmp(value, escape_char, orig_escape_len)) {
+			fprintf(outfile, "%s%s", escape_char, escape_char);
+			value += orig_escape_len;
+#endif
+		} else if (*value < 0x20) {
```

Comment:
This code effectively disallows non-ascii entries in column names.
For my purposes I changed it to:

```
		} else if ((unsigned char)*value < 0x20) { // Convert to 0xXX form only characters from [0, 0x20] range
			if (drop_nonascii) {
				putc(' ', outfile);
				++value;
			} else {
				// escape control codes / binary data.
				fprintf(outfile, "\\x%02x", *(unsigned char*)value++);
			}
		} else {
			putc(*value++, outfile);
```